server/controllers/question.js

```diff
@@ -0,0 +1,57 @@
+const Question = require("../models/question");
+const User = require("../models/user");
+var ObjectId = require("mongoose").Types.ObjectId;
+
+module.exports.getSingleQuestion = async (req, res, next) => {
+  const { difficulty } = req.body;
+  const { user } = req.user;
+  const userId = user._id;
+  console.log("user id:", userId);
+
+  try {
+    const user = await User.findById(userId);
+    const question = await Question.findOne({
+      difficulty,
+      _id: { $nin: user.questionIds },
+    });
+
+    if (question) {
+      await User.findOneAndUpdate(
+        { _id: userId },
+        { $push: { questionIds: question._id } }
+      );
```

Comment:
Again here, you can add this to the user object and then use `user.save()` to update the DB

Reply:
yes, instead of findOneAndUpdate, as I already find the user from the db, I can push the question id to the user and save the updated user into the db. I will apply this. thank you